app/js/controllers/main.js

```diff
@@ -259,20 +260,37 @@
       $scope.savedSynthNames = $scope.profile.synthNames();
     }
 
+    function importMacro() {
+      if (!$scope.profile) return;
+
+      var macroString = window.prompt('Enter new synth sequence macro:');
+
+      var regex = /\/ac(tion)?\s+"(.*?)"\s*<wait\.\d+>/g;
+      var newSequence = [];
+      var result;
+      while (result = regex.exec(macroString)) {
+        var action = result[2];
+        for (var key in $scope.allActions) {
+          var value = $scope.allActions[key];
+          if (action === value.name || action === $translate.instant(value.name)) {
+            newSequence.push(key);
+          }
+        }
+      }
+
+      if (newSequence.length > 0) {
+        $scope.sequence = newSequence;
+      }
+    }
+
     function importSynth() {
       if (!$scope.profile) return;
 
-      var synthString = window.prompt('Enter new synth sequence code:', synthString);
+      var synthString = window.prompt('Enter new synth sequence code:');
       var newSequence = JSON.parse(synthString);
```

Comment:
I didn't notice this in the last PR, but I think it would be a good idea to have some error handling here for malformed synth strings. For example, if you past a string which is missing the closing bracket, it silently does nothing except print an error stacktrace to the console.